packages/standard-minifier-js/plugin/minify-js.js

```diff
@@ -63,37 +63,43 @@ class MeteorMinifier {
       stats: Object.create(null)
     };
 
-    files.forEach(file => {
+    files.forEach((file) => {
       // Don't reminify *.min.js.
       if (/\.min\.js$/.test(file.getPathInBundle())) {
         toBeAdded.data += file.getContentsAsString();
+        toBeAdded.data += '\n\n';
+        Plugin.nudge();
       }
       else {
         let minified;
-        try {
-          minified = meteorJsMinify(file.getContentsAsString());
-        }
-        catch (err) {
+        // Async Implementation
+        meteorJsMinifyAsync(file.getContentsAsString()).then(min => {
```

Comment:
The `forEach` here won't wait for these promises.

Reply:
yep. now it is being called in order, but if there is a big file in the middle probably will change the order. I'm inverting the logic to deal with the promises after all are completed.